libcudacxx/include/cuda/std/detail/libcxx/include/memory

```diff
@@ -5010,7 +4887,7 @@ struct _LIBCUDACXX_TYPE_VIS pointer_safety
 _LIBCUDACXX_FUNC_VIS pointer_safety get_pointer_safety() noexcept;
 #else
 // This function is only offered in C++03 under ABI v1.
-# if !defined(_LIBCUDACXX_ABI_POINTER_SAFETY_ENUM_TYPE) || !defined(_LIBCUDACXX_CXX03_LANG)
```

Comment:
I believe that is not correct. This was previously always true because we would not define `_LIBCUDACXX_CXX03_LANG` so the || would make the whole expression true

Reply:
Right.
PS: Doesn't the comment above contradicts the `#ifdef`? It says that the function is only offered in C++03, but the `#if` enables it if it is not C++03 (or not `(_LIBCUDACXX_ABI_POINTER_SAFETY_ENUM_TYPE`).

Reply:
Yeah the comment is super bogus. I assume this comes from the good old double negation thing libc++ has going

Reply:
Ok, then `code` > `comment`. I will remove the ifdef altogether; thus the code will be enabled on all compile paths.